flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -570,7 +570,7 @@ private static CPUResource getCpuCores(final Configuration config, double fallba
 			if (cpuCores < 0) {
 				throw new IllegalConfigurationException("Configured cpu cores must be non-negative.");
 			}
-		} else if (fallback >= 0.0) {
```

Comment:
Looking at the other code in the if branch above this one, it looks as if `0` would be a valid `CPU_CORES` value. If we want to keep it consistent one needs to touch this branch as well or we change the default of `MESOS_RM_TASKS_CPUS` to a negative value.

Maybe the reason to allow `0` CPUs is to accept only slot requests which don't require cores in their `ResourceSpec`. But maybe @xintongsong can tell us more why `TaskExecutorResourceUtils#getCpuCores` allows to return `0` cores.

Reply:
Thanks for the review @tillrohrmann . True, changing the default of MESOS_RM_TASKS_CPUS to a negative value is another way. However, I don't get the meaning of requiring 0 cores. It seems to be an invalid value. Could @xintongsong kindly help to clarify this?

Reply:
I think the reason we allowed `TaskExecutorResourceUtils#getCpuCores` to return 0, is simply because 0 is a valid value for `CPUResource`. And the reason `CPUResource` accepts 0 is, as @tillrohrmann said, to allow slot requests to require no cores in their `ResourceSpec`.

However, for a task executor rather than a slot request, it does not make much sense to request 0 cores. Besides, after a double checking with the previous code, I believe not allowing 0 cpu cores for a task executor is more aligned with the previous behavior (for mesos, it does not make much difference for yarn/standalone).

Thus, I would agree to make `TaskExecutorResourceUtils#getCpuCores` to not allow 0 cpu cores.

Reply:
Thanks for the explanation @xintongsong . +1 to now allow 0 cpu cores.

Reply:
All right, but then we should also update the other branches of this if statement. More concretely, we need to also throw an `IllegalConfigurationException` if `cpuCores == 0`. And we should check that `config.getInteger(TaskManagerOptions.NUM_TASK_SLOTS)` returns `> 0`.